src/main/kotlin/app/extractors/DMExtractor.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Sourcerer Inc. All Rights Reserved.
+// Author: Nickolas Gupton (nickolas@gupton.xyz)
+
+package app.extractors
+
+import app.model.CommitStats
+import app.model.DiffFile
+
+class DMExtractor : ExtractorInterface {
+    companion object {
+        const val LANGUAGE_NAME = Lang.DM
+    }
+
+    override fun extractLibStats(files: List<DiffFile>): List<CommitStats> {
+        val codeExtension = ".dm"
+        val mapExtension = ".dmm"
+        val spriteExtension = ".dmi"
+
+        val codeFiles = files.filter { it.path.endsWith(codeExtension) }
+        val mapFiles = files.filter { it.path.endsWith(mapExtension) }
+        val spriteFiles = files.filter { it.path.endsWith(spriteExtension) }
+        
+
+        // Add stats from *.dm files.
+        val codeStats = listOf(CommitStats(
```

Comment:
Do not process *.dm in this section. Mapping to byond technology will be added on server.